service/operator/service.go

```diff
@@ -27,37 +30,35 @@ type Config struct {
 // DefaultConfig provides a default configuration to create a new service by
 // best effort.
 func DefaultConfig() Config {
-	var err error
-
-	var k8sClient kubernetes.Interface
-	{
-		config := k8s.DefaultConfig()
-		k8sClient, err = k8s.NewClient(config)
-		if err != nil {
-			panic(err)
-		}
-	}
-
-	var newLogger micrologger.Logger
-	{
-		config := micrologger.DefaultConfig()
-		newLogger, err = micrologger.New(config)
-		if err != nil {
-			panic(err)
-		}
-	}
-
 	return Config{
 		// Dependencies.
-		K8sClient: k8sClient,
-		Logger:    newLogger,
+		BackOff:   nil,
+		K8sClient: nil,
+		Logger:    nil,
 		Resources: nil,
 	}
 }
 
+// Service implements the service.
+type Service struct {
+	// Dependencies.
+	backOff           backoff.BackOff
+	logger            micrologger.Logger
+	operatorFramework *framework.Framework
+	resources         []framework.Resource
+
+	// Internals.
+	bootOnce sync.Once
+	mutex    sync.Mutex
+	tpr      *tpr.TPR
+}
```

Comment:
I sneaked this here in as well. The structure should be above the constructor. Thanks for bearing with me. <3